js/colourpicker.js

```diff
@@ -544,52 +655,375 @@
+    function flip(o) {
```

Comment:
one-line description of what this function does? (Though I get it from looking at the use case)